Source/EasyNetQ/Consumer/DefaultConsumerErrorStrategy.cs

```diff
@@ -200,12 +210,20 @@ private string CreateConnectionCheckMessage()
         }
 
         private bool disposed = false;
+        private bool disposing = false;
 
         public virtual void Dispose()
         {
             if (disposed) return;
+            disposing = true;
 
-            if(connection != null) connection.Dispose();
+            lock (syncLock)
```

Comment:
probably we can't do so many different things if we want to create a pure thread-safe disposable class